src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadInstallType.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.NET.Sdk.WorkloadManifestReader
+{
+    /// <summary>
+    /// Describes different workload installation types.
+    /// </summary>
+    public enum InstallType
+    {
+        /// <summary>
+        /// Workloads are installed as NuGet packages
+        /// </summary>
+        FileBased = 0,
+        /// <summary>
+        /// Workloads are installed as MSIs.
+        /// </summary>
+        Msi = 1
+    }
+
+    public static class WorkloadInstallType
+    {
+        /// <summary>
+        /// Determines the <see cref="InstallType"/> associated with a specific SDK version.
+        /// </summary>
+        /// <param name="sdkFeatureBand">The SDK version to check.</param>
+        /// <returns>The <see cref="InstallType"/> associated with the SDK.</returns>
+        public static InstallType GetWorkloadInstallType(SdkFeatureBand sdkFeatureBand, string dotnetDir)
+        {
+            string installerTypePath = Path.Combine(dotnetDir, "metadata",
+                "workloads", $"{sdkFeatureBand.ToStringWithoutPrerelease()}", "installertype");
+
+            if (File.Exists(Path.Combine(installerTypePath, "msi")))
+            {
+                return InstallType.Msi;
+            }
+
+            return InstallType.FileBased;
+        }
+
+        public static string GetInstallStateFolder(SdkFeatureBand sdkFeatureBand, string dotnetDir)
+        {
+            var installType = GetWorkloadInstallType(sdkFeatureBand, dotnetDir);
+
+            if (installType == InstallType.FileBased)
+            {
+                return Path.Combine(dotnetDir, "metadata", "workloads", sdkFeatureBand.ToString(), "InstallState");
+            }
+            else if (installType == InstallType.Msi)
+            {
+                return Path.Combine(Environment.GetEnvironmentVariable("ProgramData")!, "dotnet", "workloads", sdkFeatureBand.ToString(), "InstallState");
```

Comment:
Who creates this file?

Reply:
The dotnet CLI will create it in certain cases, like when you run certain versions of `dotnet workload update`.  See here: https://github.com/dotnet/designs/blob/simplify-workload-versioning/accepted/2023/simplify-workload-versioning-implementation.md#rolling-back-and-pinning-workload-versions

Reply:
Suggest using the GetSpecialFolder APIs, (see https://github.com/dotnet/sdk/blob/587fe23396c91efa6be825a6c8eb4aabe5787fe2/src/Cli/dotnet/Installer/Windows/MsiPackageCache.cs#L86). That calls into SHGetSpecialFolderPath AFAIK. Environment variables can always be modified by the user and are less reliable

Reply:
@joeloff Thanks, fixed